packages/sources/coinbase-prime/src/transport/wallets.ts

```diff
@@ -0,0 +1,166 @@
+import { SubscriptionTransport } from '@chainlink/external-adapter-framework/transports/abstract/subscription'
+import { EndpointContext } from '@chainlink/external-adapter-framework/adapter'
+import { PoRAddress } from '@chainlink/external-adapter-framework/adapter/por'
+import { AdapterResponse, sleep } from '@chainlink/external-adapter-framework/util'
+import { BaseEndpointTypes, inputParameters } from '../endpoint/wallets'
+import { sign } from './utils'
+import { TransportDependencies } from '@chainlink/external-adapter-framework/transports'
+import { makeLogger } from '@chainlink/external-adapter-framework/util'
+import { Requester } from '@chainlink/external-adapter-framework/util/requester'
+import { calculateHttpRequestKey } from '@chainlink/external-adapter-framework/cache'
+
+const logger = makeLogger('WalletsTransport')
+
+type Wallet = {
+  id: string
+  name: string
+  symbol: string
+  type: string
+  created_at: string
+  address: string
+}
+
+export interface ResponseSchema {
+  wallets: Wallet[]
+  pagination: {
+    next_cursor: string
+    sort_direction: string
+    has_next: boolean
+  }
+}
+
+export type WalletsTransportTypes = BaseEndpointTypes
+type RequestParams = typeof inputParameters.validated
+
+export class WalletsTransport extends SubscriptionTransport<WalletsTransportTypes> {
+  settings!: WalletsTransportTypes['Settings']
+  requester!: Requester
+  endpointName!: string
+
+  async initialize(
+    dependencies: TransportDependencies<WalletsTransportTypes>,
+    adapterSettings: WalletsTransportTypes['Settings'],
+    endpointName: string,
+    transportName: string,
+  ): Promise<void> {
+    await super.initialize(dependencies, adapterSettings, endpointName, transportName)
+    this.settings = adapterSettings
+    this.requester = dependencies.requester
+    this.endpointName = endpointName
+  }
+
+  async backgroundHandler(
+    context: EndpointContext<WalletsTransportTypes>,
+    entries: RequestParams[],
+  ) {
+    await Promise.all(entries.map(async (param) => this.handleRequest(param)))
+    await sleep(context.adapterSettings.BACKGROUND_EXECUTE_MS)
+  }
+
+  async handleRequest(param: RequestParams) {
+    let response: AdapterResponse<BaseEndpointTypes['Response']>
+    try {
+      response = await this._handleRequest(param)
+    } catch (e) {
+      const errorMessage = e instanceof Error ? e.message : 'Unknown error occurred'
```

Comment:
I think many custom transports miss this, but can we log the error here ? 

```suggestion
      const errorMessage = e instanceof Error ? e.message : 'Unknown error occurred'
     logger.error(e, errorMessage)
```